### PR TITLE
feat(client): migrate to json-rpc

### LINKED ISSUE #79: Problem: `foreign_call` not supported in kwil-js
Foreign calls were added as a property on the `Schema` struct in https://github.com/kwilteam/kwil-db/pull/664.

We will need to update the type definitions and order validation in kwil-js to support.

### PR DESCRIPTION
**Pull request by KwilLuke**
The client.ts now uses kwil-db's JSONRPC endpoints (instead of REST).

BREAKING CHANGE: Kwil-JS now relies on kwil-db's JSONRPC endpoints (available in kwil-db v0.8+). From this commit on, you should only use Kwil-JS with kwil-db v0.8+.

### CODE REVIEW

File: src/api_client/client.ts [hunk 1]
```diff
@@ -209,135 +188,154 @@ export default class Client extends Api {
...
+    throw new Error(JSON.stringify(res.data) || 'Invalid JSON RPC response.');
   }
 
-  if (!selector) {
-    return { status: res.status, data: undefined };
+  if (!res.data.result) {
+    throw new Error(JSON.stringify(res.data) || 'No result in JSON RPC response.');
   }
 
   return {
```
Comment (KwilLuke) [inline src/api_client/client.ts:340]: @jchappelow, do you mind taking a look at this `checkRes` function? I tried to model similar to how you showed me [here](https://github.com/jchappelow/kwil-db/blob/9528cd89e2702be1bbcebd848d60ceaa95e8edb4/core/rpc/client/user/jsonrpc/client.go#L134-L177).

File: src/builders/payload_builder.ts [hunk 2]
```diff
@@ -371,19 +370,20 @@ Kwil Chain ID: ${tx.body.chain_id}
...
       };
       newTx.body = {
-        description: description,
+        desc: description,
         payload: bytesToBase64(tx.body.payload as Uint8Array),
-        payload_type: newTx.body.payload_type as PayloadType,
-        fee: newTx.body.fee?.toString(),
+        type: newTx.body.type as PayloadType,
+        // @ts-ignore - DO NOT MERGE WITHOUT RESOLVING - Jon is looking into why fee is unmarshalled to an integer.
+        fee: Number(newTx.body.fee),
```
Comment (jchappelow) [inline src/builders/payload_builder.ts]: Pretty sure the resolution is simple, but need to make sure this is as expected: https://github.com/kwilteam/kwil-db/pull/713

File: src/core/auth.ts [hunk 3]
```diff
@@ -49,15 +49,15 @@ export function composeAuthMsg(
...
-  const target = new URL('auth', domain);
   let msg = '';
   msg += `${domain} wants you to sign in with your account:\n`;
   msg += `\n`;
   if (authParam.statement !== '') {
     msg += `${authParam.statement}\n`;
   }
   msg += '\n';
-  msg += `URI: ${target.href}\n`;
+  // @Yaiba: Should I trust the URI provided by KGW or should I use the domain / create my own?
```
Comment (KwilLuke) [inline src/core/auth.ts:59]: @Yaiba, can you advise on this? Should I trust the URI provided by KGW or should I create my own?

Comment (Yaiba) [reply src/core/auth.ts:59]: Since we have checked that the domain matches, KGW will always return `domain + /rpc/v1`, we can trust returned URI.

**Comment by Yaiba** [comment]
Should we merge this? I have this branch tested kwilteam/kwil-db#746  in CI https://github.com/kwilteam/kwil-db/actions/runs/9161574685/job/25186730490 and passed

**Comment by jchappelow** [comment]
> Should we merge this? I have this branch tested [kwilteam/kwil-db#746](https://github.com/kwilteam/kwil-db/pull/746) in CI https://github.com/kwilteam/kwil-db/actions/runs/9161574685/job/25186730490 and passed

Just a sec, scanning through the main handler

**Review by jchappelow — APPROVED** [review]

**Comment by jchappelow** [comment]
Yeah, lgtm.  Fire when ready @KwilLuke

**Pull request merged**